client/reader/list-stream/empty.tsx

```diff
@@ -0,0 +1,64 @@
+import DOMPurify from 'dompurify';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import EmptyContent from 'calypso/components/empty-content';
+import { recordAction, recordGaEvent } from 'calypso/reader/stats';
+import { useDispatch } from 'calypso/state';
+import { recordReaderTracksEvent } from 'calypso/state/reader/analytics/actions';
+
+export default function ListEmptyContent(): JSX.Element {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const queryParams = new URLSearchParams( location.search );
+	const lastPageLink = DOMPurify.sanitize( queryParams.get( 'last_page' ) ?? '' );
+
+	useEffect( (): void => {
+		// For clean URL remove last_page query param.
+		if ( lastPageLink ) {
+			const url = new URL( window.location.href );
+
+			url.searchParams.delete( 'last_page' );
+			history.replaceState( null, '', url );
+		}
+	}, [ lastPageLink ] );
```

Comment:
Currently, we have an issue where this will not persist past a rerender of the component. I might do something along the lines of:

```suggestion
		const [ lastPageLink, setLastPageLink ] = useState< string | null >( null );

	useEffect( (): void => {
		const queryParams = new URLSearchParams( window.location.search );
		const lastPageParam = DOMPurify.sanitize( queryParams.get( 'last_page' ) ?? '' );
		setLastPageLink( lastPageParam );
		// For clean URL remove last_page query param.
		if ( lastPageParam ) {
			const url = new URL( window.location.href );
			url.searchParams.delete( 'last_page' );
			window.history.replaceState( null, '', url );
		}
	}, [] );
```
So that value would persist after any rerenders of the component.

That said, this isnt a full solution since we have to work around that other race condition i mentioned.

Reply:
Under which condition component is going to re-render?

Reply:
I suppose mostly if its triggered up the chain such as any parents rerendering, etc. We don't seem to have much within this component itself (props/hooks) that will cause it.

I can definitely force it to rerender by changing the browser width while the component is up, there is some point there that forces a re-render and you can see the button change from profile to following. Although that is not a common flow, just a more consistent way to see it happen in testing.

Outside of that, I saw it happen once or twice in testing this PR without resizing. After a few seconds, the component rerendered (for whatever reason) and the button changed from profile back to following. This I am unsure how to trigger or see more consistently in testing.

Reply:
Thanks for noticing this re-render and suggesting the fix.

Addressed in https://github.com/Automattic/wp-calypso/pull/98706/commits/a344513fd35d188b8790aad457157305a61b0e85